examples/vault-cluster-authentication-iam/main.tf

```diff
@@ -0,0 +1,205 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A VAULT CLUSTER IN GOOGLE CLOUD
+# This is an example of how to use the vault-cluster module to deploy a private Vault cluster in GCP. A private Vault
+# cluster is the recommended approach for production usage. This cluster uses Consul, running in a separate cluster, as
+# its High Availability backend.
+# ---------------------------------------------------------------------------------------------------------------------
+
+provider "google" {
+  project = "${var.gcp_project_id}"
+  region  = "${var.gcp_region}"
+}
+
+# Use Terraform 0.10.x so that we can take advantage of Terraform GCP functionality as a separate provider via
+# https://github.com/terraform-providers/terraform-provider-google
+terraform {
+  required_version = ">= 0.10.3"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATES A SUBNETWORK WITH GOOGLE API ACCESS
+# Necessary because the private clusters don't have internet access
+# But consul and vault need to make requests to the Google API
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "google_compute_subnetwork" "private_subnet_with_google_api_access" {
+  name                     = "${var.vault_cluster_name}-private-subnet-with-google-api-access"
+  private_ip_google_access = true
+  network                  = "${var.network_name}"
+  ip_cidr_range            = "10.3.0.0/16"
```

Comment:
I have a question about this. I am picking a valid cidr since it is just an example, and this works well when running tests, however it can cause issues with range overlap if tests are more than once. E.g. I merge a PR to master and at the same time I push a new commit to one my PRs. Should I pass a random number from go tests to here? Technically the tests could still fail with `10.RAND.RAND.0/24`but that's like a chance in 65K... 
Any ideas?

Reply:
Oh, interesting. Yea, I think you'll need to have Terratest pass in some random values (that must still be valid private IPs!) and hope there are enough possibilities to make clashes unlikely. I imagine you're only using 1 or 2 IPs in the subnet, so it could be something higher than `/24` to make clashes even less likely...